components/form/form-element-mixin.js

```diff
@@ -160,6 +157,18 @@ export const FormElementMixin = superclass => class extends LocalizeCoreElement(
 		return this._validity;
 	}
 
+	connectedCallback() {
```

Comment:
Just calling attention to the only differing check here. The form mixin was adding the shadowRoot listeners in its constructor, which is valid in Lit 1 but not Lit 2. Moving the listeners to the connectedCallback fixes the issue with no functionality changes.